Cross-validation sur le meilleur modèle (XGBoost)

In [2]:
import pandas as pd
import numpy as np
from xgboost import XGBClassifier
from sklearn.model_selection import cross_val_score
df = pd.read_csv("../../data/weatherAUS_ForModel.csv")

X = df.drop(columns=["RainTomorrow", "Date", "SeasonName"], errors="ignore")
y = df["RainTomorrow"]

In [6]:

neg = np.sum(y == 0)
pos = np.sum(y == 1)
ratio = neg / pos

model = XGBClassifier(
    n_estimators=300,
    learning_rate=0.1,
    max_depth=5,
    scale_pos_weight=ratio,
    random_state=42,
    eval_metric="logloss"
)

scores = cross_val_score(model, X, y, cv=5, scoring="roc_auc")

print("Scores :", scores)
print("Moyenne :", scores.mean())
print("Écart-type :", scores.std())

Scores : [0.8523096  0.82135667 0.860919   0.76143651 0.85907231]
Moyenne : 0.8310188167476689
Écart-type : 0.037601518742945345


Avant validation croisée, le modèle obtenait un ROC-AUC d’environ 0.89, ce qui pouvait refléter une estimation optimiste des performances.
Après application d’une validation croisée à 5 plis, le ROC-AUC moyen est de 0.83 avec un écart-type de 0.037, indiquant une performance globalement satisfaisante mais une variabilité modérée selon les partitions des données.
Cette analyse montre que la validation croisée permet d’obtenir une estimation plus fiable et réaliste des performances du modèle.